### DISCUSSION TITLE
How to plot the result of throats occupy (partly)?

### DISCUSSION THREAD

**Discussion by mourendong** [PMEAL/OpenPNM#2152] [Q&A]
Hello everyone,
    I try to plot the throats occupy. The results of throats occupy were in the range of 0-1. 
The results of example 'A_ordinary_percolation' were 0 or 1, which were show by 'op.topotools.plot_connections'.
But, in my results, some throats were partly occupied (i.e., 0.4, 0.5, and so on). 
How to show these results? Can you give me some advice?

**Comment by jgostick** [comment]
Could you just use the ``color_by`` argument to color the throats according to occupancy?  The other thing that comes to mind is set the alpha value (transparency) based on the fractional occupancy.

**Reply by mourendong** [reply]
Thank you for your kind reply!
I want to plot the throats occupancy likes "Battery ratio". In one throat, the occupancy value is 0.5, the plot should show half in one color, and half in another color.
The **color_by** plot the color of total throats, which can not adjust partly in the one throat.

**Reply by jgostick** [reply]
Hmmm, I'm afraid that half-and-half is not really possible, sorry.

**Reply by ma-sadeghi** [reply]
I guess this can be done in ParaView, is that correct @jgostick?

**Reply by jgostick** [reply]
Not that I know of.  Paraview can color throats with a gradient based on the values in the two pores but I'm not sure that is what is wanted.

**Reply by mourendong** [reply]
Thanks for your reply! 
Paraview can color throats, but the color is changed totally.

**Reply by ma-sadeghi** [reply]
@mourendong I think I've made it work in the past, I could be wrong though. I'll give it a try and let you know if I find a solution.

**Reply by mourendong** [reply]
Thank you for your kind help! Looking forward to the good news of your success.

**Reply by ma-sadeghi** [reply]
@mourendong Ok, so I think that you need to add separate glyphs for throats, then color them based on a proper "throat" array, ex. `throat.occupancy`, etc. This way, the occupancy in the throats is not interpolated from adjacent pores, but rather directly comes from the throat array, which contains the correct values. [Here](https://github.com/PMEAL/OpenPNM/blob/dev/examples/contrib/plotting_in_paraview.rst)'s a tutorial on how to add glyphs for throats as it's a bit complex and unintuitive (although very straightforward!).
